src/lazy/encoder/mod.rs

```diff
@@ -0,0 +1,565 @@
+        )?;
+        Ok(self)
+    }
+
+    /// Writes any pending data to the output stream and then calls [`Write::flush`] on it.
+    pub fn flush(&mut self) -> IonResult<()> {
+        self.output.flush()?;
+        Ok(())
+    }
+
+    /// Helper method to construct this format's `ValueWriter` implementation.
+    #[inline]
+    fn value_writer(&mut self) -> TextValueWriter_1_0<'_, W> {
+        TextValueWriter_1_0 {
+            writer: self,
+            depth: 0,
+        }
+    }
+
+    /// Helper method to construct this format's `AnnotatedValueWriter` implementation.
+    #[inline]
+    fn annotated_value_writer(&mut self) -> TextAnnotatedValueWriter_1_0<'_, W> {
+        TextAnnotatedValueWriter_1_0 {
+            value_writer: self.value_writer(),
+        }
+    }
+
+    #[inline]
+    pub fn list_writer(&mut self) -> IonResult<TextListWriter_1_0<'_, W>> {
+        self.value_writer().list_writer()
+    }
+}
+
+impl<W: Write> LazyRawWriter<W, TextEncoding_1_0> for LazyRawTextWriter_1_0<W> {
+    fn new(output: W) -> Self {
+        LazyRawTextWriter_1_0::new(output)
+    }
+    // Delegate the trait methods to the inherent methods; this allows a version of these
+    // methods to be called on the concrete type even when the trait is not in scope.
+    delegate! {
+        to self {
+            fn write<V: WriteAsIon>(&mut self, value: V) -> IonResult<&mut Self>;
+            fn value_writer(&mut self) -> TextValueWriter_1_0<'_, W>;
+            fn flush(&mut self) -> IonResult<()>;
+        }
+    }
+}
+
+pub struct TextValueWriter_1_0<'a, W: Write> {
+    writer: &'a mut LazyRawTextWriter_1_0<W>,
+    depth: usize,
+}
+
+impl<'a, W: Write> TextValueWriter_1_0<'a, W> {
+    fn output(&mut self) -> &mut W {
+        &mut self.writer.output
+    }
+
+    fn whitespace_config(&self) -> &WhitespaceConfig {
+        self.writer.whitespace_config
+    }
+}
+
+pub struct TextAnnotatedValueWriter_1_0<'a, W: Write> {
+    value_writer: TextValueWriter_1_0<'a, W>,
+}
+
+impl<'a, W: Write> AnnotatedValueWriter<'a, W, TextEncoding_1_0>
+    for TextAnnotatedValueWriter_1_0<'a, W>
+{
+    fn write_annotations<
+        SymbolType: AsRawSymbolTokenRef,
+        IterType: Iterator<Item = SymbolType> + Clone,
+    >(
+        self,
+        annotations: IterType,
+    ) -> IonResult<TextValueWriter_1_0<'a, W>> {
+        let output = &mut self.value_writer.writer.output;
+        let annotations = annotations.into_iter();
+        for annotation in annotations {
+            match annotation.as_raw_symbol_token_ref() {
+                RawSymbolTokenRef::Text(token) => write!(output, "{}::", token.as_ref()),
+                RawSymbolTokenRef::SymbolId(sid) => write!(output, "${sid}::"),
+            }?;
+        }
+        Ok(self.value_writer)
+    }
+
+    fn no_annotations(self) -> TextValueWriter_1_0<'a, W> {
+        self.value_writer
+    }
+}
+
+impl<'a, W: Write> ValueWriter<'a, W, TextEncoding_1_0> for TextValueWriter_1_0<'a, W> {
```

Comment:
🗺️ The bodies of these methods are either copy/pasted from the existing raw text writer or invoke one of its methods.